database/versions/22e0b6f6ad9f_add_failed_states.py

```diff
@@ -0,0 +1,34 @@
+"""add FAILED states to checksum and validation
+
+Revision ID: 22e0b6f6ad9f
+Revises: 2c6910ed8cf6
+Create Date: 2018-10-08 16:17:17.276168
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '22e0b6f6ad9f'
+down_revision = '2c6910ed8cf6'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute('COMMIT')
+    op.execute("ALTER TYPE checksumming_status_enum ADD VALUE 'FAILED';")
+    op.execute("ALTER TYPE validation_event_status_enum ADD VALUE 'FAILED';")
+    op.add_column('validation', sa.Column('docker_image', sa.String(), nullable=True))
+    op.add_column('validation', sa.Column('original_validation_id', sa.String(), nullable=True))
+    op.execute("UPDATE validation SET status = 'FAILED' WHERE status = 'SCHEDULED';")
+    op.execute("UPDATE validation SET status = 'FAILED' WHERE status = 'VALIDATING';")
+    op.execute("UPDATE checksum SET status = 'FAILED' WHERE status = 'SCHEDULED';")
+    op.execute("UPDATE checksum SET status = 'FAILED' WHERE status = 'CHECKSUMMING';")
```

Comment:
do we want this update in the db migration? Would it be better to just run it on the db once the db includes the FAILED status?

Reply:
Hmmm I'm not sure I follow. It's good practice in my opinion to bake any data updates into the migration. This way they run without manual intervention in the correct order (add enum value and then update).  Whats the benefit of moving it out and manually running it?

Reply:
It's good to bake in the data migration only if you can unsure that the system is quiet when the migration is run.

Reply:
Ah I saw @sampierson's comment after posting mine. Thats a legitimate reason. I will remove the queries and run as part of promotion process.

Reply:
I have decided that it is best to keep this as part of the migration. We should utilize `uploadctl runlevel stop` prior to the deployment. I'll add this step to the deployment documentation